crates/ruff_linter/src/rules/flake8_simplify/rules/needless_bool.rs

```diff
@@ -70,7 +71,33 @@ pub(crate) fn needless_bool(checker: &mut Checker, stmt_if: &ast::StmtIf) {
         range: _,
     } = stmt_if;
 
-    // Extract an `if` or `elif` (that returns) followed by an else (that returns the same value)
+    let stmt_if_into_stmt = stmt_if.clone().into();
+    let next_stmt = {
+        if checker.settings.preview.is_enabled() {
+            // Get the next statement after the if statement.
+            checker
+                .semantic()
+                .current_statement_parent()
```

Comment:
If possible... we should try to only do this if we know the `if_test` matches the condition we care about.

Reply:
Good point - refactored to move this block into the match arm.